envelope.md

```diff
@@ -52,14 +52,52 @@ envelopes with individual signatures.
 }
 ```
 
+### Signature Extensions
+
+In addition to `keyid` and `sig`, a signature object may include an `extension`
+field to store ecosystem specific information pertaining to the signature.
+Extensions do not modify the signing workflow established in the
+[DSSE protocol](protocol.md). However, signers and verifiers may exchange
+information about the extensions used out-of-band.
+
+```json
+{
+  "payload": "<Base64(SERIALIZED_BODY)>",
+  "payloadType": "<PAYLOAD_TYPE>",
+  "signatures": [{
+    "keyid": "<KEYID>",
+    "sig": "<Base64(SIGNATURE)>",
+    "extension": {
+      "kind": "<EXTENSION_KIND>",
+      "ext": {...}
+    }
+  }]
+}
+```
+
+`extension.kind` is a string and the `EXTENSION_KIND` value must unambiguously
+identify the type of the extension. This in turn identifies the fields in the
+opaque object `ext`. Some well-known extension types MAY be registered and
+listed with their `ext` definition alongside the DSSE specification.
+
+Additionally, extensions must not contain any information such that the
+signature verification fails in its presence and passes in its absence.
+Essentially, if a required extension in some context is missing or if a consumer
+does not recognize the extension, verification must fail closed.
```

Comment:
This makes sense, but what is the implication here for a distributed setting?  It seems like clients without an extension may fail things which clients with the extension may pass.  

Is this intentional?  What does this mean for things like TUF or in-toto which may have may different parties performing verification of the same information?

Reply:
Yes, for example, if an in-toto client doesn't understand the sigstore extension, it must fail. This is where the producer <-> consumer agreement comes into play to ensure the signing ecosystem is used by the must-pass verifier clients.

Reply:
Does this need any edits to the doc to resolve?